### PR TITLE
Test: Improving speculos Tests + adding new currencies

### PR DESCRIPTION
**Pull request by valber-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [x] **Covered by automatic tests.** 
- [x] **Impact of the changes:** 
  - Speculos tests

### 📝 Description

Improve existing speculos tests:
- adding new coins
- adding new checks
- updating variables names

### ❓ Context

- **JIRA or GitHub link**: https://ledgerhq.atlassian.net/jira/software/c/projects/QAA/boards/887?selectedIssue=QAA-72 

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/tests/userdata/speculos.json
Comment (gre) [inline apps/ledger-live-desktop/tests/userdata/speculos.json:1]: is this an app.json? could we have a more meaningful name, like maybe the number of accounts there are?

Comment (valber-ledger) [reply apps/ledger-live-desktop/tests/userdata/speculos.json:1]: Yes, it's the app.json that is used for all the speculos tests, that's why there are a lot of accounts on it.
We are using speculos with 15 different coins (top 10 coin market cap + some coins asked by the manuals QA = total of 15 different currencies)

Comment (valber-ledger) [reply apps/ledger-live-desktop/tests/userdata/speculos.json:1]: Just changed the app.json name :)

File: apps/ledger-live-desktop/tests/specs/speculos/delete.account.spec.ts [hunk 1]
```diff
@@ -5,28 +5,31 @@ import { Application } from "tests/page";
...
+      userdata: "speculos-tests-app",
       testName: `deleteAccount_${account.currency.uiName}`,
       speculosCurrency: specs[account.currency.deviceLabel.replace(/ /g, "_")],
       speculosOffset: i,
     });
 
     //@TmsLink("B2CQA-320")
 
-    test(`[${account.currency.uiName}] Receive`, async ({ page }) => {
+    test(`[${account.currency.uiName}] Delete Account`, async ({ page }) => {
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/delete.account.spec.ts:32]: on my side only 4 test passed and 8 failed with `{"stack":"Error: Date.parse is not a function\`
image

File: apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts [hunk 2]
```diff
@@ -8,26 +8,21 @@ import { Application } from "tests/page";
...
       speculosCurrency: specs[transaction.accountToDebit.currency.deviceLabel.replace(/ /g, "_")],
       speculosOffset: i,
     });
 
-    //@TmsLink("B2CQA-479")
-    //@TmsLink("B2CQA-1904")
-    //@TmsLink("B2CQA-473")
-    //@TmsLink("B2CQA-472")
-    //@TmsLink("B2CQA-475")
+    //@TmsLink("TODO")
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts:25]: when these links will be added?

File: apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts [hunk 3]
```diff
@@ -8,26 +8,21 @@ import { Application } from "tests/page";
 const transactions = [
   //TODO: Reactivate when fees will be stable
   //new Transaction(Account.tBTC_1, Account.tBTC_2, "0.00001", "medium"),
-  //TODO: Reactivate after blobs limit in txpool is full (-32000) resolved - JIRA: BACK-7135
-  //new Transaction(Account.tETH_1, Account.tETH_2, "0.00001", "medium"),
+  new Transaction(Account.tETH_1, Account.tETH_2.address, "0.00001", "medium"),
```
Comment (ypolishchuk-ledger) [inline apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts:11]: does it make sense to test two Ethereum testnet currencies? Seems like an overkill to me

Comment (valber-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts:11]: Yes, your right, going to remove one of them

File: apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts (hunk 2 above)
Comment (valber-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts:25]: Going to work on this with this ticket: https://ledgerhq.atlassian.net/jira/software/c/projects/QAA/boards/887?assignee=712020%3A1787baec-823a-432f-bf1c-e8198ea89ef6&selectedIssue=QAA-84

File: apps/ledger-live-desktop/tests/page/account.page.ts [hunk 4]
```diff
@@ -83,9 +85,15 @@ export class AccountPage extends AppPage {
     await this.settingsButton.waitFor({ state: "visible" });
   }
 
+  @step("Expect account to be not null")
+  async expectAccountBalance() {
+    await expect(this.accountBalance).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (valber-ledger) [reply apps/ledger-live-desktop/tests/page/account.page.ts:90]: 👍

File: apps/ledger-live-desktop/tests/specs/speculos/delete.account.spec.ts (hunk 1 above)
Comment (valber-ledger) [reply apps/ledger-live-desktop/tests/specs/speculos/delete.account.spec.ts:32]: That's really weird, I just run the tests and everything is green 🤔 
Screenshot 2024-06-03 at 17 07 12

**Review by ypolishchuk-ledger — APPROVED** [review]

**Pull request merged**